src/util/loader.ts

```diff
@@ -1,23 +1,30 @@
-import fs from 'node:fs/promises';
 import path from 'node:path';
 import { pathToFileURL } from 'node:url';
 // eslint-disable-next-line import/order -- Modules in @types are handled differently
 import { load as esmLoad } from '@esbuild-kit/esm-loader';
-import yaml from 'js-yaml';
 import { require } from '../util/require.js';
 import { LoaderError } from './errors.js';
-import { loadJSON } from './fs.js';
+import { loadJSON, loadYAML } from './fs.js';
 import { timerify } from './performance.js';
 
 const load = async (filePath: string) => {
   try {
     const ext = path.extname(filePath);
-    if (ext === '.json' || ext === '.jsonc' || /rc$/.test(filePath)) {
+    if (/rc$/.test(filePath)) {
+      try {
+        // Note: `await` is needed for the try-catch to work.
+        return await loadYAML(filePath);
+      } catch {
+        return loadJSON(filePath);
```

Comment:
It would probably be better to load the file only once for both the yaml and json parsers rather than loading it twice.